cluster.go

```diff
@@ -514,13 +516,21 @@ retry:
 		return newErrResult(err)
 	}
 	resp = cc.Do(ctx, cmd)
+	if resp.Error() == errConnExpired {
+		goto retry
+	}
 process:
 	switch addr, mode := c.shouldRefreshRetry(resp.Error(), ctx); mode {
 	case RedirectMove:
-		resp = c.redirectOrNew(addr, cc, cmd.Slot(), mode).Do(ctx, cmd)
+		ncc := c.redirectOrNew(addr, cc, cmd.Slot(), mode)
+		resp = ncc.Do(ctx, cmd)
+		if resp.Error() == errConnExpired {
+			goto retry
+		}
 		goto process
 	case RedirectAsk:
-		results := c.redirectOrNew(addr, cc, cmd.Slot(), mode).DoMulti(ctx, cmds.AskingCmd, cmd)
+		ncc := c.redirectOrNew(addr, cc, cmd.Slot(), mode)
+		results := c.doMulti(ctx, ncc, cmds.AskingCmd, cmd)
```

Comment:
`ASKING` can't be handled this way. It must be sent in every retry.

Reply:
@rueian Does it mean that `ASKING` needs at head of cmds when retrying due to `errConnExpired`

Reply:
Yes, `ASKING` is only effective in the same connection and applied only to the next command; therefore, we need to re-send `ASKING` before the target `cmd` if `errConnExpired` happens.

Reply:
I see. I will also fix `askingMulti` and `askingMultiCache` as same.

Reply:
@rueian Fix [ce19729](https://github.com/redis/rueidis/pull/833/commits/ce197291cd94c3c6c61358d19a4ca131acf25592) [be811cc](https://github.com/redis/rueidis/pull/833/commits/be811cc271cf6556292e0ab81606c978f7f3fb7d)